### DISCUSSION TITLE
ASE discussion

### DISCUSSION THREAD

**Discussion by Disinterpreter** [multitheftauto/mtasa-blue#3468] [Ideas] [answered]
Current masterlist and ASE protocol are old. I think we should add new rows for every servers. But for adding the rows, we need to change the ASE protocol.

**PAY YOUR ATTENTION. THIS SUGGESTION WILL BROKE COMPATIBILITY.**

What we have at the moment:

- game
- ip
- port
- server name
- game type
- map name
- version
- password flag
- players count
- players max

What I suggest to add:

- ipv6 compatibility (see #1962)
- preferred languages (some players want to filter servers by the language)
- custom user tags (like race, roleplay, freeroam, voice)
- color(? some servers has specific color scheme it could be useful)
- discord link(?)
- email(?)

What I suggest to change:
1. change binary protocol for something flexible (like json, grpc?). Internet speed today is pretty fast and our frontend server has gzip, no need to economy bytes.
2. potential to extension, we would make some another flags like official servers (for public testing of new major versions), fork servers (for legal forked versions of the game). 

Second point of my suggestion I want to discuss here.

Appendix
https://master.mtasa.com/ase/mta/

**Comment by Fernando-A-Rocha** [comment, marked as answer]
everything you suggested except ipv6 compatibility can be solved with #3761 and #3788

without breaking compatibility, we can include the key-value pairs ("rules") that ASE contains and is already in place.

-> https://wiki.multitheftauto.com/wiki/setRuleValue

**Reply by Fernando-A-Rocha** [reply]
![image](https://github.com/user-attachments/assets/bff4de05-67df-457f-8832-38f5f1223e36)
